packages/web-worker/src/test/e2e.test.ts

```diff
@@ -520,6 +520,231 @@ describe('web-worker', () => {
       expect(page.workers()).toHaveLength(0);
     });
   });
+
+  it('terminates the worker from the main thread', async () => {
+    await withContext('terminate', async context => {
+      const {workspace, browser} = context;
+
+      await workspace.write(
+        mainFile,
+        `
+        import {createWorker, terminate} from '@shopify/web-worker';
+        self.worker = createWorker(() => import('./worker'))();
+
+        (async () => {
+          await terminate(self.worker);
+        })();
+      `,
+      );
+
+      await workspace.write(
+        workerFile,
+        `
+        export function greet() {
+          return 'Hi, friend!';
+        }
+      `,
+      );
+
+      await runWebpack(context);
+
+      const page = await browser.go();
```

Comment:
Personally I'd rather see an action that terminates (so we can show before there was a worker, after there was not)